src/integration.spec.ts

```diff
@@ -190,9 +156,100 @@ describe("pow", () => {
         await waitForBalance(pow, res1.address, bal)
       })
     })
+
+    describe("data", () => {
```

Comment:
Been keeping this integration test file up to date. It is a very thorough test of all the endpoints.